src/main/haskell/kore/src/Kore/Step/Function/Matcher.hs

```diff
@@ -454,6 +464,13 @@ matchVariableFunction
                     ]
 matchVariableFunction _ _ _ _ = nothing
 
+data ArbitrarySymbols a = ArbitrarySymbols
```

Comment:
Does "arbitrary" here mean "not constructor-like"?

Reply:
Well... Yes. Or almost, it means not constructor-modulo-like. I renamed it.